station-worldgen-api-v0/src/main/java/net/modificationstation/stationapi/api/worldgen/biome/BiomeBuilder.java

```diff
@@ -0,0 +1,202 @@
+package net.modificationstation.stationapi.api.worldgen.biome;
+
+import it.unimi.dsi.fastutil.objects.Reference2IntOpenHashMap;
+import net.minecraft.level.biome.Biome;
+import net.modificationstation.stationapi.api.worldgen.surface.SurfaceRule;
+import net.modificationstation.stationapi.impl.worldgen.BiomeColorsImpl;
+
+import javax.swing.text.html.parser.Entity;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class BiomeBuilder {
+	private static final ThreadLocal<BiomeBuilder> INSTANCES = new ThreadLocal<>();
+	private final Map<Class<? extends Entity>, Integer> hostileEntities = new Reference2IntOpenHashMap<>(32);
+	private final Map<Class<? extends Entity>, Integer> passiveEntities = new Reference2IntOpenHashMap<>(32);
+	private final Map<Class<? extends Entity>, Integer> waterEntities = new Reference2IntOpenHashMap<>(32);
```

Comment:
The type of the fields is still a usual `Map`, which hides the `Reference2IntMap#put` method that doesn't box the int.
Make the fields use `Reference2IntMap<Class<? extends Entity>>` type, and this will be resolved.

Reply:
`put` is overridden in `Reference2IntOpenHashMap`, which means that it will be used directly

Reply:
No, these are two different methods, because Integer and int are different types.
 
![image](https://github.com/ModificationStation/StationAPI/assets/19170211/604fc60c-9fc1-4a0b-9dcf-406c1f70e333)

Reply:
Now should be fixed